### PR TITLE
First version of the mutual TLS interface

### PR DESCRIPTION
**Pull request by gruyaume**
### Overview
Some charms require knowledge of other charms' public TLS certificates for trusted communication with them. This library can be used by any charm claiming to be able to authenticate (or be authenticated) through TLS communication.

### Reference
- [Mutual TLS charm relation interface](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/mutual_tls/v0).

### CODE REVIEW

File: lib/charms/mutual_tls_interface/v0/mutual_tls.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+            relation.data[self.model.unit].pop("chain")
+            certificate_removed = True
+
+        if certificate_removed:
+            logger.warning("Certificate removed from relation data")
+        else:
+            logger.warning("Can't remove certificate - No certificate in relation data")
+
+
+class MutualTLSRequires(Object):
```
Comment (ghislainbourgeois) [inline lib/charms/mutual_tls_interface/v0/mutual_tls.py:284]:
```suggestion
class MutualTLSRequires:
```

File: lib/charms/mutual_tls_interface/v0/mutual_tls.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        charm: CharmBase,
+        relationship_name: str,
+    ):
+        """Generates/use private key and observes relation changed event.
+
+        Args:
+            charm: Charm object
+            relationship_name: Juju relation name
+        """
+        super().__init__(charm, relationship_name)
```
Comment (ghislainbourgeois) [inline lib/charms/mutual_tls_interface/v0/mutual_tls.py:300]:
```suggestion
```

**Review by ghislainbourgeois — APPROVED** [review]
Some small nitpicks, but LGTM

File: lib/charms/mutual_tls_interface/v0/mutual_tls.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+## Getting Started
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.mutual_tls_interface.v0.mutual_tls
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- jsonschema
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply lib/charms/mutual_tls_interface/v0/mutual_tls.py]: Done

**Pull request merged**